### ISSUE TITLE
Where is the source code?

### ISSUE THREAD

**Issue by matthargett** [Microsoft/WinAppDriver#84]
We want to use Appium to test our React Native Windows application, which we will be porting and deploying to Windows 7, 8.1, and 10. The vast majority (~80%) of our enterprise-level, revenue-generating user base is on Windows 7.

We'd like to contribute changes to at least being WinAppDriver to Windows 8.1, but the source code isn't available on GitHub. What is the timeline to having buildable source code available in this GitHub project?

**Comment by yodurr** [comment]
Hi Matt, there's currently not a target date for opening the WinAppDriver source.  It is something we have evaluated with interest and will continue to do so.  

In the meantime please upvote your request for Win7 and 8.1 here:
https://wpdev.uservoice.com/forums/110705-universal-windows-platform/suggestions/13518501-winappdriver-support-to-win-7-and-8

Thanks.

**Comment by matthargett** [comment]
Why is there not a target date for releasing source code? Is it using private DDK headers? If there is no plan to release source code, why was this project put on GitHub? Should we just reimplement this functionality in a new project that is actually open source and supports more platforms? 

Please advise.

Thanks!

**Comment by penguinho** [comment]
I you are looking for support for older windows apps, there exists a version here that is open source.
https://github.com/2gis/Winium

It would be a small amount of work to write an appium integration for it, though without one you just launch the winium.exe and talk to it just like appium.

**Comment by matthargett** [comment]
@penguinho thanks for the pointer! @shanselman said they're working on open sourcing it ASAP:
https://twitter.com/shanselman/status/804734588866002944

... but maybe I can use Winium project to get started and hedge my bets while writing similar test code.

**Comment by fasterfish** [comment]
This issue is still relevant. Sure, support for older platforms(as windows 7 etc.) will attract quite a few end users and increase the popularity of the instrument.

**Comment by technoscavenger** [comment]
@yodurr , any update on open sourcing WinAppDriver? I am running into stability issues (WinAppDriver crashing), and I was hoping that I can help debug the issue, not easy without the source code.

**Comment by KitoCoding** [comment]
I would also like to see WinAppDriver open source. With Selenium 4.0 moving forward with W3C stardard protocols, Apium moving forward, and WinAppDriver being left int he dust, i am forced to use Selenium 3.0.1 which is not ideal.

**Comment by bhaas** [comment]
yes please - I want to fix a issue in WinAppDriver: At the moment only US keyboard layout is supported ... make the source code available on github

**Comment by kfrajtak** [comment]
I have open sourced it :) (rolled my own driver)

**Comment by ti82** [comment]
I agree that it would be nice to view the source. We are evaluating test frameworks and find that the client/server nature is much slower (twice the runtime) than using a direct out of process framework like FlaUI. I like to use MS products and tools but it's hard to argue when the speed difference is so great. I was hoping to study the client/server side of things but the source is closed!

**Comment by matthargett** [comment]
FWIW, at a previous job I ended up decompiling, patching, and reassembling the binary so we could use it on Windows 8.1 (it only supports UWP by default). Similar to MRT in .NET Native, it basically “just works” with older OSes. I didn’t have the same performance issues you mention, but we were only doing quick smoke tests on each screen of our React Native Windows app.

**Comment by jonnymoo** [comment]
Really struggling with the non US keyboard support. Opening up the source would give chance for the community to correct it.
